File: dp-ui/src/api/announcement.ts
import request from '@/api/request'

export interface Announcement {
  id?: number
  title: string
  content: string
  type: 'info' | 'success' | 'warning' | 'error'
  priority: number
  status: number
  isTop: number
  startTime?: string
  endTime?: string
  createBy?: string
  createTime?: string
  updateTime?: string
}

// 分页查询公告
export const getAnnouncementList = (params: {
  page: number
  pageSize: number
  keyword?: string
  status?: number
}) => {
  return request.get('/announcement/list', { params })
}

// 获取当前有效的公告
export const getActiveAnnouncements = () => {
  return request.get('/announcement/active')
}

// 获取公告详情
export const getAnnouncementById = (id: number) => {
  return request.get(`/announcement/${id}`)
}

// 创建公告
export const createAnnouncement = (data: Announcement) => {
  return request.post('/announcement', data)
}

// 更新公告
export const updateAnnouncement = (id: number, data: Announcement) => {
  return request.put(`/announcement/${id}`, data)
}

// 删除公告
export const deleteAnnouncement = (id: number) => {
  return request.delete(`/announcement/${id}`)
}

// 批量删除公告
export const batchDeleteAnnouncements = (ids: number[]) => {
  return request.delete('/announcement/batch', { data: ids })
}

// 获取公告统计数据
export const getAnnouncementStats = () => {
  return request.get('/announcement/stats')
}


// 更新公告状态
export const updateAnnouncementStatus = (id: number, status: number) => {
  return request.put(`/announcement/${id}/status`, null, { params: { status } })
}

File: dp-core/src/main/java/com/dataplatform/data/entity/Announcement.java
package com.dataplatform.data.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 公告实体
 */
@Data
public class Announcement {
    
    private Long id;
    
    /** 公告标题 */
    private String title;
    
    /** 公告内容 */
    private String content;
    
    /** 公告类型: info/success/warning/error */
    private String type;
    
    /** 优先级: 1-低 2-中 3-高 */
    private Integer priority;
    
    /** 状态: 0-禁用 1-启用 */
    private Integer status;
    
    /** 是否置顶: 0-否 1-是 */
    private Integer isTop;

    /** 发布范围: all-全员 dept-部门 role-角色 */
    private String targetType;

    /** 目标ID列表(JSON数组，部门或角色ID) */
    private String targetIds;

    /** 已读人数 */
    private Integer readCount;

    /** 附件JSON */
    private String attachments;
    
    /** 开始时间 */
    private LocalDateTime startTime;
    
    /** 结束时间 */
    private LocalDateTime endTime;
    
    /** 创建人 */
    private String createBy;
    
    /** 创建时间 */
    private LocalDateTime createTime;
    
    /** 更新时间 */
    private LocalDateTime updateTime;
}
